brave/brave-ios
# Fix #3632: Refactor theme logic to use iOS 13 APIs

## Summary of Changes

Fixes #3632

Theme related bugs affected:

- Fixes #3383
- Fixes #2053
- Fixes #1862
- Fixes #1722
- Fixes #1721
- Fixes #1572
- Fixes #2925 
- Fixes #1506
- Fixes #3180

## Submitter Checklist:


## Test Plan:

- Change dark mode on all parts of the UI to ensure things switch back and forth properly
- Verify you can still force a specific theme via settings and that all UI only shows colors for that theme
- Verify private mode toolbars are purple instead of the regular dark/light colors

**Known Issue**: The toolbar that appears above the keyboard when you focus a text box in a web page does not theme correctly. This is true even in the live version of the app, and seems to be an iOS specific bug. Haven't tested to determine which versions of iOS are affected.

## Screenshots:

There are many screenshots so they've been zipped into separate folders:

- [new-light.zip](https://github.com/brave/brave-ios/files/6437545/new-light.zip)
- [new-dark.zip](https://github.com/brave/brave-ios/files/6437546/new-dark.zip)
- [new-private.zip](https://github.com/brave/brave-ios/files/6437547/new-private.zip)

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
